src/components/job-definition-row.tsx

```diff
@@ -34,13 +39,78 @@ function ScheduleSummary(props: {
   return <>{cronstrue.toString(props.schedule)}</>;
 }
 
+function PauseButton(props: {
+  jobDef: Scheduler.IDescribeJobDefinition;
+  clickHandler: () => void;
+}): JSX.Element | null {
+  const trans = useTranslator('jupyterlab');
+
+  if (!props.jobDef.active) {
+    return null;
+  }
+
+  const buttonTitle = props.jobDef.name
+    ? trans.__('Pause "%1"', props.jobDef.name)
+    : trans.__('Pause job definition');
+
+  return (
+    <IconButton onClick={props.clickHandler} title={buttonTitle}>
+      <PauseIcon fontSize="small" />
+    </IconButton>
+  );
+}
+
+function ResumeButton(props: {
+  jobDef: Scheduler.IDescribeJobDefinition;
+  clickHandler: () => void;
+}): JSX.Element | null {
+  const trans = useTranslator('jupyterlab');
+
+  if (props.jobDef.active) {
+    return null;
+  }
+
+  const buttonTitle = props.jobDef.name
+    ? trans.__('Resume "%1"', props.jobDef.name)
+    : trans.__('Resume job definition');
+
+  return (
+    <IconButton onClick={props.clickHandler} title={buttonTitle}>
+      <PlayArrowIcon fontSize="small" />
+    </IconButton>
+  );
+}
+
+function DeleteButton(props: {
+  jobDef: Scheduler.IDescribeJobDefinition;
+  clickHandler: () => void;
+}): JSX.Element | null {
+  const trans = useTranslator('jupyterlab');
+
+  const buttonTitle = props.jobDef.name
+    ? trans.__('Delete "%1"', props.jobDef.name)
+    : trans.__('Delete job definition');
+
+  return (
+    <IconButton onClick={props.clickHandler} title={buttonTitle}>
+      <CloseIcon fontSize="small" />
+    </IconButton>
+  );
+}
+
 export function buildJobDefinitionRow(
   jobDef: Scheduler.IDescribeJobDefinition,
   app: JupyterFrontEnd,
-  openJobDefinitionDetail: (jobDefId: string) => unknown
+  openJobDefinitionDetail: (jobDefId: string) => unknown,
+  deleteRow: (
+    id: Scheduler.IDescribeJobDefinition['job_definition_id']
+  ) => void,
+  forceReload: () => void
 ): JSX.Element {
   const trans = useTranslator('jupyterlab');
 
+  const ss = new SchedulerService({});
```

Comment:
1. It's invalid to call a React hook outside of a React component. Although the type signature of this function is identical (e.g. `(...args: any[]) => JSX.Element`), this is still not being used like a stateful React component. The translator really should be passed in by the parent component as an argument. 

2. Because this function is called every render, the scheduler service instance is also being initialized every render. Ideally the scheduler service instance would be passed to the builder as an argument as well.

I know that the "row builder" pattern is ugly and leads to the "parameter overload" we're seeing; I had intended to refactor this later but more pressing concerns emerged. Keep in mind that there's nothing really wrong with this (i.e. it doesn't significantly increase our risk of runtime bugs) other than the fact it's ugly, so the refactor is pretty low priority ATM.

Reply:
I had previously wrapped this in a `useMemo` hook, but the whole page rendered as blank and I got a `Rendered more hooks than during the previous render` error. I can make the instances be parameters instead.